spec/requests/api/v1/npq_applications_spec.rb

```diff
@@ -167,4 +165,97 @@
       end
     end
   end
+
+  describe "POST /api/v1/npq-applications/:id/reject" do
+    let(:npq_profile) { create(:npq_validation_data, npq_lead_provider: npq_lead_provider) }
+
+    before do
+      default_headers[:Authorization] = bearer_token
+    end
+
+    it "update lead_provider_approval_status to rejected" do
+      expect { post "/api/v1/npq-applications/#{npq_profile.id}/reject" }
+        .to change { npq_profile.reload.lead_provider_approval_status }.from("pending").to("rejected")
+    end
+
+    it "responds with 200 and representation of the resource" do
+      post "/api/v1/npq-applications/#{npq_profile.id}/reject"
+
+      expect(response).to be_successful
+
+      expect(parsed_response.dig("data", "attributes", "status")).to eql("rejected")
+    end
+
+    context "application has been accepted" do
+      let(:npq_profile) { create(:npq_validation_data, npq_lead_provider: npq_lead_provider, lead_provider_approval_status: "accepted") }
+
+      it "return 400 bad request " do
+        post "/api/v1/npq-applications/#{npq_profile.id}/reject"
+        expect(response.status).to eql(400)
+      end
+
+      it "returns error in repsonse" do
+        post "/api/v1/npq-applications/#{npq_profile.id}/reject"
+
+        expect(parsed_response.key?("errors")).to be_truthy
+
+        expect(parsed_response["errors"][0].key?("title")).to be_truthy
+        expect(parsed_response["errors"][0].key?("detail")).to be_truthy
+        expect(parsed_response["errors"][0]["title"]).to eql("Status is invalid")
+        expect(parsed_response["errors"][0]["detail"]).to eql("Once accepted an application cannot change state")
+      end
+    end
+  end
+
+  describe "POST /api/v1/npq-applications/:id/accept" do
+    let(:npq_profile) { create(:npq_validation_data, npq_lead_provider: npq_lead_provider) }
+    let(:user) { npq_profile.user }
+
+    before do
+      default_headers[:Authorization] = bearer_token
+    end
+
+    it "update status to accepted" do
+      expect { post "/api/v1/npq-applications/#{npq_profile.id}/accept" }
+        .to change { npq_profile.reload.lead_provider_approval_status }.from("pending").to("accepted")
+    end
+
+    it "responds with 200 and representation of the resource" do
+      post "/api/v1/npq-applications/#{npq_profile.id}/accept"
+
+      expect(response).to be_successful
+
+      expect(parsed_response.dig("data", "attributes", "status")).to eql("accepted")
+    end
+
+    context "when participant has applied for multiple NPQs" do
+      let!(:other_npq_profile) { create(:npq_validation_data, npq_lead_provider: npq_lead_provider, user: user) }
+
+      it "rejects all other NPQs" do
+        post "/api/v1/npq-applications/#{npq_profile.id}/accept"
+
+        expect(other_npq_profile.reload.lead_provider_approval_status).to eql("rejected")
+      end
+    end
+
+    context "application has been rejected" do
```

Comment:
Is it possible to apply to another once you have an accepted application? If so, it would be good to see a test where we 400 when one of the other applications is accepted

Reply:
Yes it is possible but It would actually start causing problems. We think the chances of this happening are low. If the problem does arise we'll have investigate by hand and resolve with an offline process then reconcile after that by hand. Once we start figuring out cohorts it should reduce the problem over time but it'll still be a problem

Reply:
having now thought about this there once something is set it should not be able to change. i will make sure that is the case

Reply:
i've added an additional test (turns out it was already implemented) where once an application is accepted it cannot then be rejected by accepting a future application.

> Is it possible to apply to another once you have an accepted application? If so, it would be good to see a test where we 400 when one of the other applications is accepted

I think i misinterpreted this and was thinking of another issue. The answer is yes but I don't see why we would 400 is one of the other applications is accepted?

Reply:
Because presumably you can't have two accepted applications?

Reply:
i think we'd have have take this an edge case and deal with it offline at a later date